src/shared/user-record.c

```diff
@@ -21,6 +23,98 @@
 #define DEFAULT_RATELIMIT_BURST 30
 #define DEFAULT_RATELIMIT_INTERVAL_USEC (1*USEC_PER_MINUTE)
 
+#if ENABLE_COMPAT_MUTABLE_UID_BOUNDARIES
+static int parse_alloc_uid(const char *path, const char *name, const char *t, uid_t *ret_uid) {
+        uid_t uid;
+        int r;
+
+        r = parse_uid(t, &uid);
+        if (r < 0)
+                return log_debug_errno(r, "%s: failed to parse %s %s, ignoring: %m", path, name, t);
+        else if (uid == 0)
+                return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "%s: %s == 0, ignoring.", path, name);
```

Comment:
i still think uid == 0 is actually fine, and should not result in any log message. it might be OK to upgrade it to 1 implicitly, simply to optimize the alloc argument a bit, but as mentioned i think saying 0…999 is the system user range is totally OK, hence 0 shouldn't be refused here, even if we know that 0 is going to be root anyway... (also see earlier comment on this)